rocAL/source/loaders/numpy_source_evaluator.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "loaders/numpy_source_evaluator.h"
+
+#include "readers/image/reader_factory.h"
+
+void NumpySourceEvaluator::create(ReaderConfig reader_cfg) {
+    _reader = create_reader(std::move(reader_cfg));
+    find_max_numpy_dimensions();
+}
+
+void NumpySourceEvaluator::find_max_numpy_dimensions() {
+    _reader->reset();
+
+    while (_reader->count_items()) {
+        size_t fsize = _reader->open();
```

Comment:
Will there be files with 0 size in the data_set? Otherwise this check can be removed or can be checked in get_numby_header_data

Reply:
For numpy files whose headers cannot be parsed, we return 0 in open() function. We check for the filesize here and skip over those files in the reader